src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -159,6 +164,7 @@ public virtual async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsyn
                 .GroupBy(p => p.PackageIdentity)
                 .Select(g => g.OrderBy(p => p.TargetFramework, frameworkSorter).First());
 
+            PackageReferenceProject.CounterfactualEmitter.EmitIfNeeded(); // Emit at least one event per VS session, but not so many events
```

Comment:
```suggestion
            PackageReferenceProject.CounterfactualEmitter.EmitIfNeeded(); // Emit only one event per VS session
```

This will emit exactly one event and no more than one, right?

Reply:
yes